File: EduSpace_BE/conversation-service/src/main/java/com/eduspace/conversationservice/model/dto/response/ConversationResponse.java
package com.eduspace.conversationservice.model.dto.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class ConversationResponse {
    String conversationId;
    String conversationName;
    @JsonProperty("isActive")
    Boolean isActive;
    @JsonProperty("isAdminConversation")
    Boolean isAdminConversation;
    @JsonProperty("videoCallEnabled")
    Boolean videoCallEnabled;
    int totalMessageCount;
    int callHistoryCount;
    LocalDateTime lastActivity;
    LocalDateTime createdAt;
    @JsonProperty("isBlocked")
    Boolean isBlocked;
    @JsonProperty("isBlockedByMe")
    Boolean isBlockedByMe;
    int unreadCount;
    String lastMessage;
    OtherUser otherUser;

    public LocalDateTime getLastActivityCompat() {
        return lastActivity;
    }

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    @FieldDefaults(level = AccessLevel.PRIVATE)
    public static class OtherUser {
        String userId; // keycloakId
        String fullName;
        String email;
        String avatarUrl;

        public String getUserId() { return userId; }
        public void setUserId(String userId) { this.userId = userId; }
        public String getFullName() { return fullName; }
        public void setFullName(String fullName) { this.fullName = fullName; }
        public String getEmail() { return email; }
        public void setEmail(String email) { this.email = email; }
        public String getAvatarUrl() { return avatarUrl; }
        public void setAvatarUrl(String avatarUrl) { this.avatarUrl = avatarUrl; }
    }

    public String getConversationId() { return conversationId; }
    public void setConversationId(String conversationId) { this.conversationId = conversationId; }
    public String getConversationName() { return conversationName; }
    public void setConversationName(String conversationName) { this.conversationName = conversationName; }
    public Boolean getIsActive() { return isActive; }
    public void setIsActive(Boolean active) { isActive = active; }
    public Boolean getIsAdminConversation() { return isAdminConversation; }
    public void setIsAdminConversation(Boolean adminConversation) { isAdminConversation = adminConversation; }
    public Boolean getVideoCallEnabled() { return videoCallEnabled; }
    public void setVideoCallEnabled(Boolean videoCallEnabled) { this.videoCallEnabled = videoCallEnabled; }
    public int getTotalMessageCount() { return totalMessageCount; }
    public void setTotalMessageCount(int totalMessageCount) { this.totalMessageCount = totalMessageCount; }
    public int getCallHistoryCount() { return callHistoryCount; }
    public void setCallHistoryCount(int callHistoryCount) { this.callHistoryCount = callHistoryCount; }
    public LocalDateTime getLastActivity() { return lastActivity; }
    public void setLastActivity(LocalDateTime lastActivity) { this.lastActivity = lastActivity; }
    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }
    public Boolean getIsBlocked() { return isBlocked; }
    public void setIsBlocked(Boolean blocked) { isBlocked = blocked; }
    public Boolean getIsBlockedByMe() { return isBlockedByMe; }
    public void setIsBlockedByMe(Boolean blockedByMe) { isBlockedByMe = blockedByMe; }
    public int getUnreadCount() { return unreadCount; }
    public void setUnreadCount(int unreadCount) { this.unreadCount = unreadCount; }
    public String getLastMessage() { return lastMessage; }
    public void setLastMessage(String lastMessage) { this.lastMessage = lastMessage; }
    public OtherUser getOtherUser() { return otherUser; }
    public void setOtherUser(OtherUser otherUser) { this.otherUser = otherUser; }
}

File: eduspace_mobile_v2/lib/data/models/response/message/conversation_response.dart
class ConversationResponse {
  final String conversationId;
  final String conversationName;
  final bool? isActive;
  final bool? isAdminConversation;
  final bool? videoCallEnabled;
  final int totalMessageCount;
  final int callHistoryCount;
  final DateTime? lastActivity;
  final DateTime? createdAt;
  final bool? isBlocked;
  final bool? isBlockedByMe;
  final int unreadCount;
  final String? lastMessage;
  final OtherUser? otherUser;

  ConversationResponse({
    required this.conversationId,
    required this.conversationName,
    this.isActive,
    this.isAdminConversation,
    this.videoCallEnabled,
    this.totalMessageCount = 0,
    this.callHistoryCount = 0,
    this.lastActivity,
    this.createdAt,
    this.isBlocked,
    this.isBlockedByMe,
    this.unreadCount = 0,
    this.lastMessage,
    this.otherUser,
  });

  factory ConversationResponse.fromJson(Map<String, dynamic> json) {
    return ConversationResponse(
      conversationId: json['conversationId'] ?? '',
      conversationName: json['conversationName'] ?? '',
      isActive: json['isActive'],
      isAdminConversation: json['isAdminConversation'],
      videoCallEnabled: json['videoCallEnabled'],
      totalMessageCount: json['totalMessageCount'] ?? 0,
      callHistoryCount: json['callHistoryCount'] ?? 0,
      lastActivity: json['lastActivity'] != null ? DateTime.parse(json['lastActivity']) : null,
      createdAt: json['createdAt'] != null ? DateTime.parse(json['createdAt']) : null,
      isBlocked: json['isBlocked'],
      isBlockedByMe: json['isBlockedByMe'],
      unreadCount: json['unreadCount'] ?? 0,
      lastMessage: json['lastMessage'],
      otherUser: json['otherUser'] != null ? OtherUser.fromJson(json['otherUser']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'conversationId': conversationId,
      'conversationName': conversationName,
      'isActive': isActive,
      'isAdminConversation': isAdminConversation,
      'videoCallEnabled': videoCallEnabled,
      'totalMessageCount': totalMessageCount,
      'callHistoryCount': callHistoryCount,
      'lastActivity': lastActivity?.toIso8601String(),
      'createdAt': createdAt?.toIso8601String(),
      'isBlocked': isBlocked,
      'isBlockedByMe': isBlockedByMe,
      'unreadCount': unreadCount,
      'lastMessage': lastMessage,
      'otherUser': otherUser?.toJson(),
    };
  }
}

class OtherUser {
  final String userId;
  final String fullName;
  final String email;
  final String? avatarUrl;

  OtherUser({
    required this.userId,
    required this.fullName,
    required this.email,
    this.avatarUrl,
  });

  factory OtherUser.fromJson(Map<String, dynamic> json) {
    return OtherUser(
      userId: json['userId'] ?? '',
      fullName: json['fullName'] ?? '',
      email: json['email'] ?? '',
      avatarUrl: json['avatarUrl'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'userId': userId,
      'fullName': fullName,
      'email': email,
      'avatarUrl': avatarUrl,
    };
  }
}
